packages/arb-tx-aggregator/batcher/queue.go

```diff
@@ -0,0 +1,262 @@
+package batcher
+
+import (
+	"container/heap"
+	"errors"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/offchainlabs/arbitrum/packages/arb-evm/message"
+	"github.com/offchainlabs/arbitrum/packages/arb-tx-aggregator/snapshot"
+	arbcommon "github.com/offchainlabs/arbitrum/packages/arb-util/common"
+	"log"
+	"math/rand"
+)
+
+// An IntHeap is a min-heap of ints.
```

Comment:
I'm confused by the comments here. The comments seem to imply that this is a priority-queue abstraction (implemented as a heap).  But the code seems to be a simple in-order queue.  I don't see any of the sorting / heapifying operations that a heap would use.

Reply:
I updated the comments a bit. The `TxHeap` struct implements the requirements from https://golang.org/pkg/container/heap/ which we use for its heap method. The comments had been copied from the code example located in that documentation